src/main/java/com/powsybl/sld/server/NetworkAreaDiagramService.java

```diff
@@ -41,12 +42,19 @@ class NetworkAreaDiagramService {
 
     public SvgAndMetadata generateNetworkAreaDiagramSvg(UUID networkUuid, String variantId, List<String> voltageLevelsIds, int depth) {
         Network network = DiagramUtils.getNetwork(networkUuid, variantId, networkStoreService, PreloadingStrategy.COLLECTION);
-        voltageLevelsIds.forEach(voltageLevelId -> {
+        AtomicInteger numberOfNotExistingVl = new AtomicInteger();
+        List<String> voltageLevelsIdsCopy = new ArrayList<>(voltageLevelsIds);
+        voltageLevelsIdsCopy.forEach(voltageLevelId -> {
             if (network.getVoltageLevel(voltageLevelId) == null) {
```

Comment:
maybe use a basic `for` loop to avoid copying the array

Reply:
done, i used iterator